chains/evm/contracts/pools/USDC/USDCTokenPoolProxy.sol

```diff
@@ -0,0 +1,308 @@
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn
+  ) internal view virtual returns (bool) {
+    // Cache the legacy pool address to avoid multiple SLOADs.
+    address legacyPool = s_pools.legacyCctpV1Pool;
+
+    // If the legacy pool without a proxy is not set, then there is no need to check the destinationCaller.
+    if (legacyPool == address(0)) {
+      return false;
+    }
+
+    USDCTokenPool.MessageAndAttestation memory msgAndAttestation =
+      abi.decode(releaseOrMintIn.offchainTokenData, (USDCTokenPool.MessageAndAttestation));
+
+    bytes32 destinationCallerBytes32;
+    bytes memory messageBytes = msgAndAttestation.message;
+    assembly {
+      // destinationCaller is a 32-byte word starting at position 84 in messageBytes body, so add 32 to skip the 1st word
+      // representing bytes length.
+      destinationCallerBytes32 := mload(add(messageBytes, 116)) // 84 + 32 = 116
+    }
+    address destinationCaller = address(uint160(uint256(destinationCallerBytes32)));
+
+    return destinationCaller == legacyPool;
+  }
+
+  /// @notice Converts a legacy sourcePoolData struct into a new format that can be used to release or mint USDC on the
+  /// previous pool. This is necessary because the sourcePoolData is stored in a different format in the previous pool,
+  /// and must be in a properly decodable format.
+  /// @param releaseOrMintIn The releaseOrMintIn struct to generate a new struct for.
+  /// @return newReleaseOrMintIn The new releaseOrMintIn struct.
+  function _generateNewReleaseOrMintIn(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn
+  ) internal view returns (Pool.ReleaseOrMintInV1 memory newReleaseOrMintIn) {
+    // Copy the releaseOrMintIn struct to the newReleaseOrMintIn struct. We do this to avoid having to copy each field
+    // individually, which would be more gas intensive, as only the sourcePoolData field is going to be modified.
+    newReleaseOrMintIn = releaseOrMintIn;
+
+    // Get the local domain identifier and the transmitter proxy address from the previous pool. The local
+    // domain identifier is not present in the original data, so it must be acquired dynamically, and simplifies this
+    // contract by not having to store it in the state.
+    uint32 localDomain = USDCTokenPool(s_pools.cctpV1Pool).i_localDomainIdentifier();
+    bytes32 allowedCaller =
+      bytes32(uint256(uint160(address(USDCTokenPool(s_pools.cctpV1Pool).i_messageTransmitterProxy()))));
+
+    // Decode the legacy sourcePoolData to get the nonce and sourceDomain. The original sourcePoolData was a struct
+    // with two fields, a uint64 and a uint32. We can decode it into two variables directly, nonce and sourceDomain.
+    (uint64 nonce, uint32 sourceDomain) = abi.decode(releaseOrMintIn.sourcePoolData, (uint64, uint32));
+
+    // Create the new payload out of the legacy sourcePoolData.
+    USDCTokenPool.SourceTokenDataPayload memory newPayload = USDCTokenPool.SourceTokenDataPayload({
```

Comment:
can inline to avoid assingment

Reply:
https://github.com/smartcontractkit/chainlink-ccip/pull/1134/commits/e6a274ec773c5d2540f10f3733a988d1f7d47001#diff-7d40f3f52c8e578caab476a14e440188e432d8f912bfa3f6871375cc86ac5e4eR272